src/index.js

```diff
@@ -1,26 +1,36 @@
-import React from "react";
+import React, {Component, PropTypes} from "react";
 
-class ReactImageFallback extends React.Component {
+class ReactImageFallback extends Component {
 	constructor(props) {
 		super(props);
 		this.state = {
 			displayImage: props.initialImage
 		};
+		this.setImage = this.setImage.bind(this);
 	}
 
-	componentDidMount() {
+	setImage(props) {
 		let displayImage = new Image();
 		displayImage.onerror = () => {
 			this.setState({
-				displayImage: this.props.fallbackImage
+				displayImage: props.fallbackImage
 			});
 		};
 		displayImage.onload = () => {
 			this.setState({
-				displayImage: this.props.src
+				displayImage: props.src
 			});
 		};
-		displayImage.src = this.props.src;
+		displayImage.src = props.src;
+	}
+	componentDidMount() {
+		this.setImage(this.props);
+	}
+
+	componentWillReceiveProps(nextProps) {
+		if (nextProps.shouldUpdateImage(this.props, nextProps)) {
```

Comment:
yeah, the space images have the same url even when they update. I haven't figured out how i will decide to update it but i know it can't be as simple as that. I also like the idea of having more power over the component. I do see some merit it defaulting to that though.